apps/portal/app/buckets/[address]/_components/bucket.tsx

```diff
@@ -120,15 +119,34 @@ export default function Bucket({ bucketAddress }: { bucketAddress: Address }) {
                   </Link>
                 </BreadcrumbLink>
               ) : (
-                displayAddress(bucketAddress)
+                <div className="text-foreground flex items-center gap-2 font-semibold">
+                  {displayAddress(bucketAddress)}
+                  <CopyButton
+                    value={bucketAddress}
+                    tooltip="Copy bucket ID"
+                    successMessage="Bucket ID copied to clipboard"
+                    className="opacity-40 hover:opacity-100"
+                  />
+                </div>
               )}
             </BreadcrumbItem>
             {!!pathParts.length && <BreadcrumbSeparator />}
             {pathParts.map((part, index) => (
               <Fragment key={`${index}-${part}`}>
                 <BreadcrumbItem>
                   {index === pathParts.length - 1 ? (
-                    part || "\u00A0\u00A0"
+                    <div className="text-foreground flex items-center gap-2 font-semibold">
+                      <File className="text-primary mr-1 size-4" />
+                      {part || "\u00A0\u00A0"}
+                      <CopyButton
+                        value={part}
+                        tooltip={
+                          isObject ? "Copy filename" : "Copy directory name"
+                        }
+                        successMessage={`${isObject ? "Filename" : "Directory name"} copied to clipboard`}
+                        className="opacity-40 hover:opacity-100"
+                      />
```

Comment:
Was still imaging the copy button would live after the `Breadcrumb` component down below here. You could remove this `CopyButton` and the one above, and just have a single `CopyButton` down below after the closure of the `Breadcrumb` component. The value passed into `CopyButton` would depend on if you're viewing the bucket root (the bucket id) or a folder/file (the path). If you do that, you an also remove the `<div>` currently surrounding each `CopyButton`.

Reply:
okay. played with this. struggled with line wrapping. if on its own it has its own wrapping and linebreak behavior. i could fix the linebreak behavior but wrapping remained an issue. ideally it would only wrap _with_ the last element of the breadcrumb, that's just not what was happening. perhaps improving the breadcrumb could be a follow-up ticket

Reply:
besides if this is okay, i think the rest is gtg